src/nautilus-window.c

```diff
@@ -527,9 +543,36 @@ const GActionEntry win_entries[] = {
 	{ "prompt-root-location", action_prompt_for_location_root },
 	{ "prompt-home-location", action_prompt_for_location_home },
 	{ "go-to-tab", NULL, "i", "0", action_go_to_tab },
+	{ "custom-key-press", NULL, "s", "''", action_custom_key_press },
 };
 
 static void
+nautilus_load_custom_shortcuts(GApplication *app){ 
+	gchar *path = g_strconcat(g_get_home_dir(),SHORTCUTS_PATH,NULL);
+	if ( access(path, R_OK) != -1 ) {
```

Comment:
You can use `g_file_get_contents()` and `g_strsplit()` on newlines to load the accelerator map and iterate it line-by-line.